zrml/prediction-markets/src/migrations.rs

```diff
@@ -19,6 +19,288 @@
+            total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+
+            for id in bounded_vec.into_inner().iter() {
+                if let Ok(market) = <zrml_market_commons::Pallet<T>>::market(id) {
+                    if market.dispute_mechanism == MarketDisputeMechanism::Authorized {
+                        authorized_resolutions.insert(*id, resolve_at);
+                    }
+                } else {
+                    log::warn!("AddFieldToAuthorityReport: Could not find market with id {:?}", id);
+                }
+            }
+        }
+
+        let mut new_storage_map: Vec<(
+            <T as zrml_market_commons::Config>::MarketId,
+            AuthorityReport<BlockNumberFor<T>>,
+        )> = Vec::new();
+
+        for (market_id, old_value) in storage_key_iter::<
+            <T as zrml_market_commons::Config>::MarketId,
+            OutcomeReport,
+            Twox64Concat,
+        >(AUTHORIZED, AUTHORIZED_OUTCOME_REPORTS)
+        {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+
+            let resolve_at: Option<BlockNumberFor<T>> =
+                authorized_resolutions.get(&market_id).cloned();
+
+            match resolve_at {
+                Some(block) => {
+                    new_storage_map.push((
+                        market_id,
+                        AuthorityReport { resolve_at: block, outcome: old_value },
+                    ));
+                }
+                None => {
+                    log::warn!(
+                        "AddFieldToAuthorityReport: Market was not found in \
+                         MarketIdsPerDisputeBlock; market id: {:?}",
+                        market_id
+                    );
+                    // example case market id 432
+                    // https://github.com/zeitgeistpm/zeitgeist/pull/701 market id 432 is invalid, because of zero-division error in the past
+                    // we have to handle manually here, because MarketIdsPerDisputeBlock does not contain 432
+                    new_storage_map.push((
+                        market_id,
+                        AuthorityReport {
+                            resolve_at: <BlockNumberFor<T>>::zero(),
+                            outcome: old_value,
+                        },
+                    ));
+                }
+            }
+        }
+
+        for (market_id, new_value) in new_storage_map {
+            let hash = utility::key_to_hash::<
+                Twox64Concat,
+                <T as zrml_market_commons::Config>::MarketId,
+            >(market_id);
+            put_storage_value::<AuthorityReport<T::BlockNumber>>(
+                AUTHORIZED,
+                AUTHORIZED_OUTCOME_REPORTS,
+                &hash,
+                new_value,
+            );
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        }
+
```

Comment:
Yeah, the last review was about that. We decided to leave the auto resolutions as they are.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/862#discussion_r1045256664

Reply:
I can't find the right conversation at the moment about this. So just let me explain it again.

In my last commits I decided to delete all `resolution blocks` for `authorized` like you mentioned. But then we had the following problem: What if the authority already made its report. Then we never resolve.

> Less important, but still something to keep an eye on is that the "old" reports are migrated in a fashion that is close to what would have happend to them if they had been submitted after these changes.

Yeah, so the best would be to have the same behaviour as before for authorized markets. Let me think about that, how this would impact the storage migration and behaviour...

Then I just left the way the resolutions are currently.

Reply:
This means, that authorized resolves as expected, but can resolve earlier if you authorize a market outcome again.

Reply:
But why would we keep the auto-resolve in?

Reply:
Effort analysis. We have no item in `MarketIdsPerDisputeBlock`. It is just overkill to make an effort. When we delete all authorized `MarketIdsPerDisputeBlock` items, we would miss the case, that the authority already made its report and we have no resolution then. We would need to recognise the case, that the authority already made its report and so on. It is just complex as hell for the benefit we get.

Reply:
Sure, leave it the way it is. One suggestion though: If you use the current block instead of block zero for the broken markets, this gives us a chance to use Battery Station's sudo key to make a correction shortly after the migration. If you set it to zero, we don't get that window of opportunity.

Reply:
> If you use the current block instead of block zero for the broken markets

I don't set a block to zero. What do you exactly mean?

Reply:
```rust
        pub fn authorize_market_outcome(
            origin: OriginFor<T>,
            market_id: MarketIdOf<T>,
            outcome: OutcomeReport,
        ) -> DispatchResultWithPostInfo {
            T::AuthorizedDisputeResolutionOrigin::ensure_origin(origin)?;
            let market = T::MarketCommons::market(&market_id)?;
            ensure!(market.status == MarketStatus::Disputed, Error::<T>::MarketIsNotDisputed);
            ensure!(market.matches_outcome_report(&outcome), Error::<T>::OutcomeMismatch);
            ensure!(
                market.dispute_mechanism == MarketDisputeMechanism::Authorized,
                Error::<T>::MarketDoesNotHaveDisputeMechanismAuthorized
            );

            let mut ids_len: Option<u32> = None;
            let report = match AuthorizedOutcomeReports::<T>::get(market_id) {
                Some(report) => AuthorityReport { resolve_at: report.resolve_at, outcome },
                None => {
                    let now = frame_system::Pallet::<T>::block_number();
                    let correction_period_ends_at = now.saturating_add(T::CorrectionPeriod::get());
                    ids_len = Some(T::DisputeResolution::add_auto_resolve(
                        &market_id,
                        correction_period_ends_at,
                    )?);
                    AuthorityReport { resolve_at: correction_period_ends_at, outcome }
                }
            };

            AuthorizedOutcomeReports::<T>::insert(market_id, report);

            if let Some(len) = ids_len {
                Ok(Some(T::WeightInfo::authorize_market_outcome_first_report(len)).into())
            } else {
                Ok(Some(T::WeightInfo::authorize_market_outcome_existing_report()).into())
            }
        }
```

You mean the migration, which sets the blocknumber of the authority report to zero (for the market 432)?

If you mean this, there is no item in `MarketIdsPerDisputeBlock` and even, when you got zero inside `resolve_at` in `AuthorityReport`, then you don't resolve as long as there is no item in `MarketIdsPerDisputeBlock`. So you have all time in the world to authorize the market outcome of this market.

Reply:
7a6eaad7110b2208addd96a534a3692817df3778

Reply:
c67a4a1ced20571bf78f5fe9022c8c9c9f72c58c